test/boost/conftest.py

```diff
@@ -0,0 +1,261 @@
+#
+# Copyright (C) 2024-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+import asyncio
+import io
+import logging
+import os
+import subprocess
+import sys
+import time
+from pathlib import Path, PosixPath
+from typing import Dict, Sequence, Union, Optional
+from xml.etree import ElementTree
+
+import pytest
+from _pytest.nodes import Collector
+
+from test.pylib.cpp.common_cpp_conftest import collect_items, get_modes_to_run, pytest_collection_modifyitems, pytest_addoption
+from test.pylib.cpp.facade import TestFacade, CppTestFailure, run_process
+from test.pylib.host_registry import HostRegistry
+from test.pylib.minio_server import MinioServer
+from test.pylib.s3_server_mock import MockS3Server
+from test.pylib.util import LogPrefixAdapter
+
+TIMEOUT_DEBUG = 60 * 5
+TIMEOUT = 60 * 2
+
+def pytest_collect_file(file_path: PosixPath, parent: Collector):
+    """
+    Method triggered automatically by pytest to collect files from a directory. Boost and unit have the same logic for
+    collection, the only difference in execution, and it's covered by facade
+    """
+    return collect_items(file_path, parent, facade=BoostTestFacade(parent.config), run_id=parent.config.getoption('run_id'))
+
+
+@pytest.hookimpl(wrapper=True)
+def pytest_runtestloop(session):
+    """
+    https://docs.pytest.org/en/stable/reference/reference.html#pytest.hookspec.pytest_runtestloop
+    This hook is needed to start the Minio and S3 mock servers before tests. After starting the servers, the default
+    pytest's runtestloop takes control. Finally part is responsible for stopping servers regardless of failure in the tests.
+    """
+    modes = get_modes_to_run(session)
+    is_worker = False
+    if "xdist" in sys.modules:
+        import xdist
+        is_worker = xdist.is_xdist_worker(session)
```

Comment:
Wdyt about using `sys.modules["xdist"]` instead of importing it? For me it looks little bit more consistent.